### DISCUSSION TITLE
Welcome to WhatsappWeb4j Discussions!

### DISCUSSION THREAD

**Discussion by Auties00** [Auties00/WhatsappWeb4j#25] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 475059538** [comment]
Hi, Mr Autiero

I'm a Java developer from China. 
I'm working on task of communicating with whatsapp web api, so I just wonder whether your project whatsappweb4j is good for that task. I'v read all document on git page, but can't run project rightly
so, here are my questions: 
1. whether this project is good for real  whatsapp web communication, not just a framework demo
2. how to config or login my whatsapp account,
3. because of network restrictions in china, is there necessary and how to config proxy

**Reply by Auties00** [reply]
Hello,
a proxy is probably needed. Do you want me to add it as a feature?

**Comment by Daligz** [comment]
Hello @Auties00 , I recently used this incredible library again but I get the following error, my code worked well before.
https://pastebin.com/M4i0s3vn

**Reply by Auties00** [reply]
Hello, I didn't see your message until now. Is it fixed? If not please open an issue

**Comment by EliyahuStern** [comment]
Hi!

Thanks for this great repository!
I've started experimenting with the provided API after struggling with the protocol on my own for a while.
A few things I would like to know:

Can I send messages to phone numbers not in my contacts? An attempt I made to create a Chat from a jid and displayName (do they always look like "<international_phone_number>@s.whatsapp.net"?) worked, but I don't have numbers who are not in my contact list to play with.
Is it possible to work with multiple instances of the API/connection in parallel? It seems like the credentials are always stored in the same place in Preferences. I think there's a pending PR related. Am I right?
Can this library be also used for connecting as the main application, or just Whatsappweb? I mean, can I generate keys from a received SMS as if I am registering to Whatsapp on my mobile, instead of by scanning a QR code?
Your help is really appreciated. Of course, I would be glad to contribute to the above features (and others) if needed.

**Reply by Auties00** [reply]
So you can message whoever you want. You just have to create a new contact object from your jid using the provided constructor. Multi instance is not supported as of now, it is coming though in the md branch. The API also supports only WhatsappWeb, not Whatsapp, so you can't get an SMS
